org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/workspace/WorkspaceAccess.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.workspace;
+
+import org.eclipse.core.internal.resources.WorkManager;
+import org.eclipse.core.internal.resources.Workspace;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.runtime.CoreException;
+
+/**
+ * @author Sebastian Zarnekow - Initial contribution and API
+ */
+@SuppressWarnings("restriction")
+public interface WorkspaceAccess {
+	
+	enum Result {
+		YES,
+		NO,
+		SHUTDOWN
+	}
+
+	static Result isWorkspaceLockedByCurrentThread(IWorkspace workspace) {
```

Comment:
am not sure if i like the static here (for patchability)

Reply:
I'll make this injectable.